mqtt/mqtt-broker/src/session.rs

```diff
@@ -279,12 +279,12 @@ impl DisconnectingSession {
 
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct SessionState {
-    client_id: ClientId,
-    subscriptions: HashMap<String, Subscription>,
+    pub client_id: ClientId,
```

Comment:
Let's not increase visibility of fields. They are used in the broker logic, so no need to make them more than private. Usually what people do - implement `into_parts(self)->(T1, T2, ...)`  method to deconstruct struct. You can find examples in the broker code